app/rtc_compare.py

```diff
@@ -0,0 +1,329 @@
+                                                                        hdf5_in_2,
+                                                                        key_dataset,
+                                                                        is_attr=False)
+            if list_flag_identical_dataset[id_flag]:
+                print(f'{id_flag+1:03d} / {len(union_set_dataset):03d} : '
+                      f'PASSED. key: {key_dataset}')
+            else:
+                print(f'\033[91m{id_flag+1:03d} / {len(union_set_dataset):03d} : '
+                      f'FAILED. key: {key_dataset}\033[00m')
+
+        # Check the attribute
+        union_set_attrs = set_attrs_1.union(set_attrs_2)
+
+        if len(union_set_attrs) == len(set_attrs_1) and \
+           len(union_set_attrs) == len(set_attrs_2):
+            flag_identical_attrs_structure = True
+            print('\nAttribute structure identical.')
+
+        else:
+            flag_identical_attrs_structure = False
+            print('\nAttribute structure not identical.')
+            print('In the 1st HDF5, not in the 2nd data:')
+            print('\n'.join(list(set_dataset_1 - set_dataset_2)))
+            print('In the 2st HDF5, not in the 1nd data:')
+            print('\n'.join(list(set_dataset_2 - set_dataset_1)))
+
+        # Proceed with checking the values in dataset,
+        # regardless of the agreement of their structure.
+
+        list_flag_identical_attrs = [None] * len(union_set_attrs)
+        for id_flag, key_attr in enumerate(union_set_attrs):
+            list_flag_identical_attrs[id_flag] = comapre_dataset_attr(hdf5_in_1,
+                                                                      hdf5_in_2,
+                                                                      key_attr,
+                                                                      is_attr=True)
+            str_printout = key_attr.replace('\n',' - ')
+            if list_flag_identical_attrs[id_flag]:
+                print(f'{id_flag+1:03d} / {len(union_set_attrs):03d} : '
+                      f'PASSED. path - key: {str_printout}')
+            else:
+                print(f'\033[91m{id_flag+1:03d} / {len(union_set_attrs):03d} : '
+                      f'FAILED. path - key: {str_printout}\033[00m')
+
+
+        # Print out the test summary:
+        print('\n\n****************** Test summary ******************')
+        print(f'1st HDF FILE                 : {file_1}')
+        print(f'2nd HDF FILE                 : {file_2}')
+        print(f'Value tolerance              : {RTC_S1_PRODUCTS_ERROR_TOLERANCE}')
+        print(f'\nIdentical dataset structure  : {flag_identical_dataset_structure}')
+        print(f'Identical Attribute structure: {flag_identical_attrs_structure}\n')
+
+        if all(list_flag_identical_dataset):
+            print('All dataset passed the test')
+        else:
+            print('Dataset below did not pass the test:')
```

Comment:
This is not priority for now, but it would be nice if we could add more info why they don't pass the test

Reply:
I've added some lines to print out the information about the failed test.